zen/utils/template.py

```
@@ -1,70 +1,197 @@
+        Basic variable validation using registry metadata.
+
+        - Ensures required variables are present
+        - Provides hints for missing optional blocks
+        """
+        requirements = entry.get("requirements", {})
+        required_vars = requirements.get("required_variables", [])
+        missing = [var for var in required_vars if var not in variables]
+        if missing:
+            raise TemplateRegistryError(
+                f"Template '{entry['id']}' missing required variables: {', '.join(missing)}"
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)